app/javascript/packs/new_application.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+// $FlowIgnore[missing-export] it is exported but name-mapper is failing
+import { NewApplicationFormWrapper } from 'NewApplication'
+import { safeFromJsonString } from 'utilities/json-utils'
+
+document.addEventListener('DOMContentLoaded', () => {
+  const containerId = 'new-application-form'
+  const container = document.getElementById(containerId)
+
+  if (!container) {
+    return
+  }
+
+  const { dataset } = container
+
+  const { createServicePlanPath, createApplicationPath, createApplicationPlanPath, serviceSubscriptionsPath } = dataset
+  const product = safeFromJsonString(dataset.product)
+  const products = safeFromJsonString(dataset.products)
+  const servicePlansAllowed = safeFromJsonString(dataset.servicePlansAllowed)
+  const buyer = safeFromJsonString(dataset.buyer)
+  const buyers = safeFromJsonString(dataset.buyers)
+  const definedFields = safeFromJsonString(dataset.definedFields)
+  const validationErrors = safeFromJsonString(dataset.errors) || {}
+  const error: string | void = validationErrors.hasOwnProperty('base') ? validationErrors.base[0] : undefined
```

Comment:
Is not enough `error:  ?string` ?

Reply:
Didn't work well the first time but I'll try that again.

Reply:
OK I thought it didn't work but the problem is we don't want a maybe type here, we want either a string or undefined because optional props `error?: string` don't accept it. I've found the whole `?type` thing is prone to error and in the end it's easier to just be explicit.